src/js/components/FrameworkConfigurationForm.js

```diff
@@ -245,11 +245,26 @@ class FrameworkConfigurationForm extends Component {
     } = this.props;
 
     const TitleField = props => {
-      return (
-        <h2 className="flush-top short-bottom">
-          {StringUtil.capitalizeEveryWord(props.title)}
-        </h2>
-      );
+      const level = props.formContext.level;
```

Comment:
@dcos/frontend The code below feels odd having to check to "random selected" numbers to set the headers size.

I debugged a bit to see if there was a better approach and then discussed with @ahoskins my discomfort about it.

I'm curious to know if anyone has a better idea we can do this.

Reply:
A way we could **almost know** this is from the `idSchema`, which is passed into each schema value. IdSchema (as the name suggests) gives each node in the schema a unique string. For example "root_service_name" is `service.name` in the JSON schema. The issue is how id schema will split **multi-word keys** - it'll do "root_coordinator_nodes_name" for `coordinator_nodes.name`. So you can't just count the underscores to know the depth.

I don't understand why [idSchema](https://github.com/mozilla-services/react-jsonschema-form/blob/51d403ab448326183f58b5cc7c2c750c200f0242/src/utils.js#L622) would split by "_" rather than ".", because this would get rid of this ambiguity. 🙃 

I think the current solution is the simplest we can do, but I agree with @weblancaster that it's a bit odd.

Reply:
@ahoskins is there any context that is passed down? We should not use the `idSchema` as that is not reliable, nor a good practice :)

Reply:
Actually, I'm just noticing that there is `formContext`. @weblancaster can you point to where the randomness is so I can understand the oddity a bit better please?

Reply:
@MatApple the `level` comes from `registry.formContext.level + 1` which I think is a bit random and odd when checking for `level === 1`.

What I meant by that the code here isn't telling us anything. 
A good example on the other hand, would be something like `serviceStatus === ServiceStatus.WARNING`

Does that make sense?

If the team agree this is the best solution I agree too, just feels like this is code smell.

Reply:
I see - you're suggesting a constant would be more appropriate here instead of an assigned integer and then the mapping of headings to integers should also be a constant. Sounds reasonable to me. I am also ok with the nesting counter as is `level + 1` since we're measuring recursion depth. How that is mapped to heading elements should be easily editable as a constant.

Reply:
Agree. I will introduce constants here.

Reply:
Thanks for the changes @ahoskins 

Small change but reads so much better now.